packages/experience/src/containers/SocialSignInList/index.tsx

```diff
@@ -17,6 +18,16 @@ const SocialSignInList = ({ className, socialConnectors = [] }: Props) => {
   const { invokeSocialSignIn, theme } = useSocial();
   useNativeMessageListener();
 
+  const [loadingStates, setLoadingStates] = useState<Record<string, boolean>>(
+    Object.fromEntries(socialConnectors.map((connector) => [connector.id, false]))
+  );
+
+  const handleClick = async (connector: ExperienceSocialConnector) => {
+    setLoadingStates((previous) => ({ ...previous, [connector.id]: true }));
+    await invokeSocialSignIn(connector);
+    setLoadingStates((previous) => ({ ...previous, [connector.id]: false }));
+  };
```

Comment:
We can set the loading connector ID to the state, instead of maintaining a complex object

Reply:
Good call, updated.